### PR TITLE
Improve configuration constraint violation message

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java [hunk 1]
```diff
@@ -248,8 +248,22 @@ else if (!prefix.isEmpty()) {
         }
 
         for (ConstraintViolation<?> violation : VALIDATOR.validate(instance)) {
-            problems.addError("Constraint violation with property prefix '%s': %s %s (for class %s)",
-                    prefix, violation.getPropertyPath(), violation.getMessage(), configClass.getName());
+            String propertyFieldName = violation.getPropertyPath().toString();
+            // upper case first character to match config attribute name
```
Comment (electrum) [inline configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java]: I don't understand this. Why would anything other than a class name start with uppercase?

**Comment by electrum** [comment]
Can you add a test for this?

**Comment by electrum** [comment]
Looks good

File: configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java [hunk 2]
```diff
@@ -248,8 +248,22 @@ else if (!prefix.isEmpty()) {
...
+            String attributeName = Character.toUpperCase(propertyFieldName.charAt(0)) + propertyFieldName.substring(1);
```
Comment (johngmyers) [inline configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java]: `CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, propertyFieldName)` ?

File: configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java (hunk 1 above)
Comment (dain) [inline configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java]: This is how configuration metadata in *this* class works.

**Pull request merged**